### PR TITLE
New set

### PR DESCRIPTION
**Pull request by Daniel-GrunbergerCA**
support new creating, updating and deleting cronjob commads

### CODE REVIEW

File: mainhandler/imageregistryhandlerhelper.go [hunk 1]
```diff
@@ -2,27 +2,76 @@ package mainhandler
...
 	"github.com/golang/glog"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func (actionHandler *ActionHandler) updateRegistryScanCronJob() error {
-	jobParams := actionHandler.command.GetCronJobParams()
-	if jobParams == nil {
-		glog.Infof("updateRegistryScanCronJob: failed to get jobParams")
-		return fmt.Errorf("failed to get failed to get jobParams")
+func (actionHandler *ActionHandler) updateSecret(sessionObj *utils.SessionObj, secretName string, authConfig *types.AuthConfig) error {
```
Comment (Bezbran) [inline mainhandler/imageregistryhandlerhelper.go:19]: Do we need to change helm chart RBAC accordingly to allow us creating/updating/deleteing secrets?
Or we already have those permissions?

File: mainhandler/imageregistryhandlerhelper.go [hunk 2]
```diff
@@ -80,5 +196,31 @@ func (actionHandler *ActionHandler) deleteRegistryScanCronJob() error {
 		return fmt.Errorf("failed to get jobParams")
 	}
 
-	return actionHandler.deleteCronjob(jobParams.JobName)
+	name := jobParams.JobName
+
+	err := actionHandler.k8sAPI.KubernetesClient.BatchV1beta1().CronJobs(armotypes.KubescapeNamespace).Delete(context.Background(), name, metav1.DeleteOptions{})
```
Comment (Bezbran) [inline mainhandler/imageregistryhandlerhelper.go:201]: batchv1betaa1 isn't dependat on the cluster?

File: mainhandler/kubescapehandler.go [hunk 3]
```diff
@@ -86,7 +86,7 @@ func (actionHandler *ActionHandler) setKubescapeCronJob() error {
 			return err
 		}
 
-		jobTemplateObj, err := getCronJobTemplate(actionHandler.k8sAPI, KubescapeCronJobTemplateName)
+		jobTemplateObj, err := getCronJobTemplate(actionHandler.k8sAPI, KubescapeCronJobTemplateName, utils.Namespace)
```
Comment (Bezbran) [inline mainhandler/kubescapehandler.go:89]: armotypes.namespace???

**Review by Bezbran — CHANGES_REQUESTED** [review]
utils.Namespace<>armotypes.Namespce???

File: mainhandler/imageregistryhandlerhelper.go (hunk 1 above)
Comment (Daniel-GrunbergerCA) [reply mainhandler/imageregistryhandlerhelper.go:19]: no, we will need to update it

File: mainhandler/imageregistryhandler.go [hunk 4]
```diff
@@ -209,31 +214,57 @@ func (rs *registryScan) filterRepositories(repos []string) []string {
 	}
 	filteredRepos := []string{}
 	for _, repo := range repos {
-		if rs.registry.projectID != "" {
-			if !strings.Contains(repo, rs.registry.projectID+"/") {
-				continue
-			}
-		}
+		// if rs.registry.projectID != "" {
```
[the remark this answers is not in the document]
Comment (Daniel-GrunbergerCA) [reply mainhandler/imageregistryhandler.go:217]: Yes. I saw that  in ecr we did not have project ID in repo name, and we were not scanning it, that's why I removed. It doesn't impact gcr

**Review by Bezbran — APPROVED** [review]

**Pull request merged**